src/modules/UI/Wallets/action.js

```diff
@@ -74,54 +73,57 @@ export const upsertWallet = (wallet: AbcCurrencyWallet) => (dispatch: any, getSt
   })
 }
 
-// setEnabledTokens is specifically for enabling them *within the GUI*, not within the core
-export const setEnabledTokens = (walletId: string, enabledTokens: any) => (dispatch: any, getState: any) => {
+// adds to core and enables in core
+export const addCustomToken = (walletId: string, tokenObj: any) => (dispatch: any, getState: any) => {
+  const state = getState()
+  const wallet = CORE_SELECTORS.getWallet(state, walletId)
+  WALLET_API.addCoreCustomToken(wallet, tokenObj)
+  .then(() => {
+    return
+  })
+  .catch((e) => console.log(e))
+}
+
+export const setEnabledTokens = (walletId: string, enabledTokens: Array<string>, disableTokens: Array<string>) => (dispatch: any, getState: any) => {
+  // tell Redux that we are updating the enabledTokens list
   dispatch(setTokensStart())
+  // get a snapshot of the state
   const state = getState()
+  // get a copy of the relevant core wallet
   const wallet = CORE_SELECTORS.getWallet(state, walletId)
-  WALLET_API.setSyncedTokens(wallet, enabledTokens)
-  .then((tokens) => {
+  // now actually tell the wallet to enable the token(s) in the core and save to file
+  WALLET_API.setEnabledTokens(wallet, enabledTokens, disableTokens)
```

Comment:
please use consistent naming, `enabledTokens` is a noun, `disableTokens` is a verb